src/media/js/addon/components/addon.js

```diff
@@ -65,46 +54,119 @@ export class Addon extends React.Component {
   }
 
   render() {
-    const disabled = this.props.isPublishing || this.props.isRejecting;
-
     return (
       <div className="addon">
         <div>
           {this.renderName()}
         </div>
         <dl>
-          <dt>Files</dt>
-          <dd>
-            <a href={this.props.unsigned_download_url}>
-              Download {this.props.name} .zip
-            </a>
-          </dd>
+          <dt>Slug</dt>
+          <dd>{this.props.slug}</dd>
 
           <dt>Manifest</dt>
           <dd>
-            <a href={this.props.manifest_url}>
-              Download {this.props.name} manifest
+            <a href={this.props.mini_manifest_url}>
+              Download manifest
             </a>
           </dd>
+        </dl>
 
-          <dt>Slug</dt>
-          <dd>{this.props.slug}</dd>
+        {!this.props.isListing &&
+          <VersionListing isReview={this.props.isReview}
+                          publish={this.props.publish}
+                          reject={this.props.reject}
+                          slug={this.props.slug}
+                          versions={versionListSelector(this.props.versions)}/>
+        }
+      </div>
+    );
+  }
+}
 
-          <dt>Status</dt>
-          <dd>{this.props.status}</dd>
 
+class VersionListing extends React.Component {
+  static propTypes = {
+    isReview: React.PropTypes.bool,
+    versions: React.PropTypes.array.isRequired,
+
+    publish: React.PropTypes.func,
+    reject: React.PropTypes.func,
+  };
+
+  render() {
+    return (
+      <div className="version-listing">
+        <h3>Versions</h3>
+
+        <ul>
+          {this.props.versions.map(version =>
+            <li>
+              <Version {...this.props} {...version}/>
+            </li>
+          )}
+        </ul>
+      </div>
+    );
+  }
+}
+
+
+class Version extends React.Component {
+  static PropTypes = {
+    id: React.PropTypes.number.isRequired,
+    download_url: React.PropTypes.string.isRequired,
+    slug: React.PropTypes.string.isRequired,
+    status: React.PropTypes.string.isRequired,
+    version: React.PropTypes.string.isRequired,
+    unsigned_download_url: React.PropTypes.string.isRequired,
+
+    // `isReview` determines whether public/reject buttons are displayed.
+    // It will be `false` in cases such as the AddonDashboard.
+    isReview: React.PropTypes.bool,
```

Comment:
Can we rename it to something a little more descriptive? `allowActions` or `allowProcessing` or something?